lib/lwm2m_carrier/os/lwm2m_os.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <lwm2m_os.h>
+
+#include <stdlib.h>
+#include <zephyr.h>
+#include <nvs/nvs.h>
+#include <logging/log.h>
+#include <secure_services.h>
+#include <misc/util.h>
+
+/* NVS-related defines */
+
+/* Multiple of FLASH_PAGE_SIZE */
+#define NVS_SECTOR_SIZE    DT_FLASH_ERASE_BLOCK_SIZE
+/* At least 2 sectors */
+#define NVS_SECTOR_COUNT   3
+/* Start address of the filesystem in flash */
+#define NVS_STORAGE_OFFSET DT_FLASH_AREA_STORAGE_OFFSET
+
+static struct nvs_fs fs = {
+	.sector_size = NVS_SECTOR_SIZE,
+	.sector_count = NVS_SECTOR_COUNT,
+	.offset = NVS_STORAGE_OFFSET,
+};
+
+int lwm2m_os_init(void)
+{
+	int err;
+
+	/* Initialize storage */
+	err = nvs_init(&fs, DT_FLASH_DEV_NAME);
```

Comment:
Just

`return nvs_init(&fs, DT_FLASH_DEV_NAME);
`

does the same and is easier to read.

Reply:
True, we had more stuff in that function so that's why it looks like that :)
I think it's fine to keep it, in case more things will need to happen during initialization.

Reply:
I disagree, it can be expanded at that time.

Right now it's communicating that it's doing more than it is.